app/views/companies/show.html.erb

```diff
@@ -24,9 +24,17 @@
                 </div>
               <% end %>
             <% unless @company.user == current_user %>
-                <a class="btn btn-flat message px-0" href="#" style="width: 201px;"><i class="far fa-heart fa-lg"></i> Add to favorites</a>
-                <a class="btn btn-flat message px-0" href="#" style="width: 201px;"><i class="fab fa-telegram-plane fa-lg"></i> Send message</a>
-            <% end %>
+              <% if current_user.developer && current_user.developer.favourites.include?(@company) %>
+                <%= link_to company_favourite_path(current_user.developer.company_favourites.find_by(company: @company)), method: :delete, class: 'btn btn-flat favorite' do %>
+                  <i class="fas fa-ban fa-lg"></i> Remove from favorites
+                <% end %>
+                <% else %>
+                  <%= link_to company_company_favourites_path(@company), method: :post, class: "btn btn-flat btn-delete w-100" do %>
+                    <i class="far fa-heart fa-lg"></i> Add to favorites
+                  <% end %>
+                <% end %>
+                  <a class="btn btn-flat btn-delete w-100" href="#"> <i class="fab fa-telegram-plane fa-lg"></i> Send message</a>
```

Comment:
If I'm logged in as a developer, it would be better to hide "add to favourites" on another developer's profile, as it would make no sense to add another dev to favourites, plus the button will never actually do anything.